README.md

```diff
@@ -1,135 +1,277 @@
-Silverstripe CMS 4.x is stored in
-[https://github.com/silverstripe/developer-docs/tree/4/en](https://github.com/silverstripe/developer-docs/tree/4/en).
+```bash
+npm ci                 # Install dependencies (use npm ci, not npm update)
 
-The userhelp documentation is stored in the [silverstripe/silverstripe-userhelp-content repo](https://github.com/silverstripe/silverstripe-userhelp-content/).
+# Development with mock data (recommended for development)
+npm run mock           # Dev server with mock data (no cloning needed)
 
-## What to update when creating a new pre-release major branch, making a stable major release, or making a major EOL
+# Development with real cloned content
+npm run dev:docs       # Dev server with developer docs on port 9876
+npm run dev:user       # Dev server with user help on port 9876
+npm run clone:docs     # Clone developer docs without starting dev server
+npm run clone:user     # Clone user help docs without starting dev server
 
-When **creating a new major branch for a pre-release major version**
+# Production build & test
+npm run build          # Build & lint (default: docs context, alias for build:docs)
+npm run build:docs     # Build developer docs
+npm run build:user     # Build user help
 
-- Make sure you've added a new major branch to both `silverstripe/developer-docs` and `silverstripe/silverstripe-userhelp-content`
-- Add the new major to `sources-docs.js` and `sources-user.js`
-- Add the new major branches for various modules as defined in `sources-docs.js` and `sources-user.js` as well
-- Add the new major to the version select in `src/components/Header.tsx`
-- Add the new major to the `PRE_RELEASE` array in the `getVersionMessage` function in `src/utils/nodes.ts`
-- Add the new major version to the [algolia crawler script](https://crawler.algolia.com/admin/crawlers/3d14ccdd-f9ae-4957-bc0a-5b21b4c29af3/configuration/edit)
-
-For **new stable releases**, you will need to do the following
+# Testing
+npm test               # Run all tests
+npm run test:watch     # Run tests in watch mode
+npm run lint           # Run ESLint (must have 0 errors, 0 warnings)
+```
 
-- Remove the major from the `PRE_RELEASE` array in the `getVersionMessage` function in `src/utils/nodes.ts`
-- Update the `getDefaultVersion` function's return value to the new stable major in `src/utils/nodes.ts`
-- Update redirects in `netlify.toml`
-- Update `src/pages/index.tsx` to navigate to the new stable major
-- Add the new major to `redirects` in `gatsby-node.js`
+---
 
-When a **major goes EOL**, add the major to the `EOL` array in the `getVersionMessage` function in `src/utils/nodes.ts`
+## Environment Setup
 
-## Installation
+### Environment Variables
 
-To set up a local instance of [doc.silverstripe.org](https://github.com/silverstripe/doc.silverstripe.org):
+Create `.env.local` for local development:
 
-* Clone this repository to an empty directory
+```bash
+# Algolia DocSearch (optional, search disabled if not set)
```

Comment:
Since this is optional, remove the "Optional Environment Variables" heading - all of the environment variables are optional it sounds like.